### ISSUE TITLE
Allow references to local JS plugins

### ISSUE THREAD

**Issue by robw-mercury** [oxc-project/oxc#20779]
In ESLint you can import a plugin module and use the reference in place of the "plugin name":

```ts
import myLocalEslintPlugin from './eslintPlugin/index.mjs'

export default [
{ 
  plugins: {
    '@myLocalPlugin': myLocalEslintPlugin,
  },
  rules: {
    '@myLocalPlugin/myRule': 'error',
  }
  // ...rest of config
}
``` 

With the Oxlint `mts` config, `jsPlugins` `specifier` must be a resolvable string from package.json. In order to use local plugins, I must package them and install them in a repo, which is over complicated.

Can `specifier` be a reference to the plugin module, or a relative path to the a plugin module file?

**Comment by Sysix** [comment]
> Can `specifier` be a reference to the plugin module, or a relative path to the a plugin module file?

Yes you can use a relative path:
https://oxc.rs/docs/guide/usage/linter/js-plugins.html#using-js-plugins

> The path can be any valid import specifier e.g. ./plugin.js, eslint-plugin-foo, or @foo/eslint-plugin. Paths are resolved relative to the config file itself.

Does this test example help you?:
https://github.com/oxc-project/oxc/tree/main/apps/oxlint/test/fixtures/js_config_js_plugins

**Comment by robw-mercury** [comment]
Apologies @Sysix, I just didn't see that doc. Thank you and appreciate the work you all are doing!
